src/bp/core/services/job-service.ts

```diff
@@ -11,11 +12,26 @@ export interface JobService {
    * @param T The return type of the returned function
    */
   broadcast<T>(fn: Function): Promise<Function>
+
+  acquireLock(resource: string, duration: number): Promise<RedisLock | undefined>
+
+  clearLock(resource: string): Promise<boolean>
 }
 
 @injectable()
 export class CEJobService implements JobService {
   async broadcast<T>(fn: Function): Promise<Function> {
     return fn
   }
+
+  async acquireLock(resource: string, duration: number): Promise<RedisLock | undefined> {
```

Comment:
Can you add a TODO here to actually implement and test it for single-node?  It will be useful in certain use-cases even if single node

Reply:
Yep sure, I guess you would like to use it to lock ghost resources across all nodes when writing files. Good idea!